opentelemetry/proto/metrics/v1/metrics.proto

```diff
@@ -95,14 +95,141 @@ message InstrumentationLibraryMetrics {
+//
+// Both TimeUnixNano and StartTimeUnixNano values are expressed as
+// UNIX Epoch time in nanoseconds since 00:00:00 UTC on 1 January 1970.
+//
+// # TimeUnixNano
+//
+// This field is required, having consistent interpretation across
+// DataPoint types.  TimeUnixNano is the moment corresponding to when
+// the data point's aggregate value was captured.
+//
+// Data points with/ the 0 value for TimeUnixNano SHOULD be rejected
+// by consumers.
+//
+// # StartTimeUnixNano
+//
+// StartTimeUnixNano in general allows detecting when a sequence of
+// observations is unbroken.  StartTimeUnixNano is set to the
+// beginning of an unbroken sequence of observations (all point kinds
+// except with delta aggregation temporality) or the end of an
+// unbroken sequence of observations (point kinds with delta
+// temporality).
+//
+// This field indicates to consumers the start time for points with
+// cumulative and delta AggregationTemporality, and it should be
+// included whenever possible to support correct rate calculation.
+// Although it may be omitted when the start time is truly unknown,
+// setting StartTimeUnixNano is strongly encouraged.
+//
+// There are two reasons that streams with AggregationTemporality are
+// reset: (1) a process restarts, and (2) the process "forgets" the stream
+// and then re-initializes it.  StartTimeUnixNano is critical for
+// calculating correct rates for Monotonic Sum, Histogram, and Summary
+// data points in both of these cases.
+//
+// Another use for this information which applies to all point kinds
+// is to prevent overlap when multiple writers accidentally write to a
+// single stream.  StartTimeUnixNano MAY be used for overlap detection
+// for all point kinds.
+//
+// ## StartTimeUnixNano: Sums and Histograms
+//
+// Since these two point kinds have AggregationTemporality, the use of
+// StartTimeUnixNano is strongly recommended to ensure correct rate
+// calculation and prevent overlap.
+//
+// ## StartTimeUnixNano: Gauges
+//
+// The use of StartTimeUnixNano is optional for Gauge points.  This is
+// recommended in cases where there is a risk of mis-interpretion caused
+// by overlapping points (by a violation of the single-writer principle).
+//
+// When used, StartTimeUnixNano should be set to the start time of the
+// process or the timestamp when the Gauge was first set.
+//
+// ## StartTimeUnixNano: Summaries
+//
+// Summary points are treated the same as for cumulative aggregation
+// temporality, by convention.  StartTimeUnixNano should be set to the
+// starting timestamp in an unbroken sequence of observations.
+//
+// # Unknown StartTimeUnixNano
+//
+// This protocol adopts the Prometheus heuristic for reset detection,
+// and with it prescribes a solution for filling in StartTimeUnixNanos
+// with as much information as possible to assist consumers with
+// calculating rates and detecting overlap.  This process can be
+// applied to Sum and Histogram points with cumulative aggregation
```

Comment:
You cannot apply that unless you know:
1. Sum is monotonic
2. Histogram Sum is monotonic (which is something that we don't know, and have issue https://github.com/open-telemetry/opentelemetry-proto/issues/187), which I think was not resolved, because we need at least to add a comment that the sum is "counter".

Reply:
**EDIT** For histograms we should use the "count" as the counter and all other bins values are counters, so we are good on that.

But the question about #187 still stays, it says that is resolved in a manner but I don't see that in the documentation.

Reply:
+1 to adding documentation on how to handle things as we discussed in #187.   Do you want that in a separate PR (I can do that) or we can update this PR with it.

Reply:
Separate PR is good.

Reply:
I have tried to separate the topic of "reset" from the topic of computing a rate. The notion of a reset I'm trying to develop has to do with eliminating overlapping series that may have been accidentally written _as well as_ calculating correct rates.

The logic here applies a StartTime to all the kinds of points you might find in a Prometheus WAL, for example, and the outcome is an OTLP stream that is not overlapping with itself _as well as_ encodes information about reset monotonics.

This formalizes the Prometheus heuristic for resetting counters in a way that lets OTLP points be written out of order, but it also ensures that two Prometheus servers configured to write the same series, converted into OTLP this way would be detected as overlapping.